arcaflow_plugin_fio/fio_schema.py

```diff
@@ -58,108 +58,357 @@ def is_sync(self) -> bool:
+            "reported separately."
+        ),
+    ] = None
+
+    # Time related parameters
+    runtime: typing.Annotated[
+        typing.Optional[str],
+        schema.name("Job Run Time"),
+        schema.description(
+            "Limit runtime. The test will run until it completes the configured I/O "
+            "workload or until it has run for this specified amount of time, whichever "
+            "occurs first. When the unit is omitted, the value is interpreted in "
+            "seconds."
+        ),
+    ] = None
+    time_based: typing.Annotated[
+        typing.Optional[int],
+        validation.min(0),
+        validation.max(1),
+        schema.name("Time Based"),
+        schema.description(
+            "If set, fio will run for the duration of the runtime specified even if "
+            "the file(s) are completely read or written. It will simply loop over the "
+            "same workload as many times as the runtime allows."
+        ),
+    ] = None
+    startdelay: typing.Annotated[
+        typing.Optional[str],
+        schema.name("Job Start Delay"),
+        schema.description(
+            "Delay the start of job for the specified amount of time. Can be a single "
+            "value or a range. When given as a range, each thread will choose a value "
+            "randomly from within the range. Value is in seconds if a unit is omitted."
+        ),
+    ] = None
+    ramp_time: typing.Annotated[
+        typing.Optional[str],
+        schema.name("Job Ramp Time"),
+        schema.description(
+            "Run the specified workload for this amount of time before logging any "
+            "performance numbers. Useful for letting performance settle before logging "
+            "results, thus minimizing the runtime required for stable results. Note "
+            "that the ramp_time is considered lead in time for a job, thus it will "
+            "increase the total runtime if a special timeout or runtime is specified. "
+            "When the unit is omitted, the value is given in seconds."
+        ),
+    ] = None
+
+    # Target file/device
+    directory: typing.Annotated[
+        typing.Optional[str],
+        schema.name("Job Directory"),
+        schema.description(
+            "Prefix filenames with this directory. Used to place files in a different "
+            "location than './'. You can specify a number of directories by separating "
+            "the names with a ':' character. These directories will be assigned "
+            "equally distributed to job clones created by numjobs as long as they are "
+            "using generated filenames. If specific filename(s) are set fio will use "
+            "the first listed directory, and thereby matching the filename semantic "
+            "(which generates a file for each clone if not specified, but lets all "
+            "clones use the same file if set)."
+        ),
+    ] = None
+    filename: typing.Annotated[
+        typing.Optional[str],
```

Comment:
I've been debating this along with the handling of booleans and some other things in the schema. The question is whether its more valuable to better manage the plugin inputs or to allow an existing fio input file or command line to be entered as normal for the underlying tool.

Reply:
Okay. One option is to have two mutually exclusive fields. One is the raw field, and one is the schema-friendly field. But this isn't required. It may be more work than you find worth it.